examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -101,6 +101,28 @@ void OTAProviderExample::SetOTAFilePath(const char * path)
 void OTAProviderExample::SetOTACandidates(std::vector<OTAProviderExample::DeviceSoftwareVersionModel> candidates)
 {
     mCandidates = std::move(candidates);
+
+    // Validate that each candidate matches the info in the image header
+    for (auto candidate : mCandidates)
+    {
+        OTAImageHeader header;
+        ParseOTAHeader(candidate.otaURL, header);
+
+        VerifyOrDie(candidate.vendorId == header.mVendorId);
```

Comment:
Please add a printout telling the user why the application failed -- maybe not for every single case but at least some info to indicate what went wrong and with which image file.